### PR TITLE
feat: dynamic routing for each event on MyEvent page

### PR DESCRIPTION
**Pull request by aidenm1**
[//]: # "Feel free to customize this template to your project's vibes"

### :croissant: Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Added dynamic routing for each event in my event page. 

### :baguette_bread: Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

image

## :sandwich: How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Review the querying functions within api/supabase/queries. Specifically, events, facility, and volunteers. (I believe these functions are good)

Next, review app/events/[event_id]/page.tsx and app/events/[event_id]/styles.ts. I'm not sure if I did the styling the conventional way, but it seems to work. 

## :pie: Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

image
In Jinkang's design, the top of the content area is curved, I'm not entirely sure how to do this.

Will phone numbers be formatted in the database, or should I create a function to format the phone numbers so that they are (xxx)xxx-xxxx?

Should second contact be either Volunteer Host or Facility Host depending on needs_host bool, or should it just be Volunteer Host?

Can we guarantee that there will always be a volunteer host who signed up and a performer who signed up? If not guaranteed, should there be a separate page rendered saying that event details haven't been finalized yet?

### :pretzel: Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### :bagel: Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @celinechoiii

### CODE REVIEW

File: app/events/[event_id]/styles.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+`;
+
+export const EventImage = styled(NextImage)`
+  width: 100%;
+  height: auto;
+  max-height: 11.5rem;
+  object-fit: cover;
+`;
+
+export const GradientOverlay = styled.div`
```
Comment (celinechoiii) [inline app/events/[event_id]/styles.ts:31]: i think you can get rid of the `GradientOverlay` styling

File: app/events/[event_id]/styles.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+export const BackImage = styled(NextImage)`
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 1.5rem;
+  height: 1.5rem;
+  margin: 1rem;
+`;
+
+export const ImageWrapper = styled.div`
```
Comment (celinechoiii) [inline app/events/[event_id]/styles.ts:17]: `ImageWrapper` can be deleted as well!

File: app/events/[event_id]/styles.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  position: relative;
+  width: 100%;
+  height: auto;
+  display: flex;
+`;
+
+export const EventImage = styled(NextImage)`
```
Comment (celinechoiii) [inline app/events/[event_id]/styles.ts:24]:
```
export const EventImage = styled(NextImage)`
  width: 100%;
  height: 23vh;
  object-fit: cover;
  position: relative;
`;
```

File: app/events/[event_id]/styles.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  width: 100%;
+  height: 100%;
+  background: linear-gradient(
+    0deg,
+    rgba(0, 0, 0, 0.35) 0%,
+    rgba(0, 0, 0, 0.35) 100%
+  );
+`;
+
+export const Page = styled.div`
```
Comment (celinechoiii) [inline app/events/[event_id]/styles.ts:44]: i set the default background to be `gray1`

```
export const Page = styled.div`
  flex-direction: column;
  min-width: 100%;
  min-height: 100svh;
  overflow: hidden;
`;
```

File: app/events/[event_id]/styles.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  background-color: ${COLORS.gray1};
+  flex-direction: column;
+  min-width: 100%;
+  min-height: 100svh;
+  overflow: hidden;
+  margin-bottom: 3.75rem;
+`;
+
+export const Container = styled.div`
```
Comment (celinechoiii) [inline app/events/[event_id]/styles.ts:52]: - `border-radius` field helps us round the corners!
- deleting `margin-bottom` field in `Page` since bottom padding is directly being added to this container
- negative margins help us overlap elements and `z-index` sets the order of overlapped elements. the default z-index value is 0 and higher z-index values bring elements to the front, while lower values push them to the back
- i'm making the background white for now so that the container doesn't blend in with the default background color
```
export const Container = styled.div`
  z-index: 10;
  margin: -1.5rem 0;
  position: relative;
  padding: 1.5rem 2rem 4rem 2rem;
  display: flex;
  flex-direction: column;
  border-radius: 20px 20px 0 0;
  background: white;
`;
```

File: app/events/[event_id]/styles.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
+'use client';
+
+import NextImage from 'next/image';
+import styled from 'styled-components';
+import COLORS from '@/styles/colors';
```
Comment (celinechoiii) [inline app/events/[event_id]/styles.ts:5]: delete this line as no global colors are being used

File: app/events/[event_id]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+    true,
+  );
+  const location = facility.street_address_1 + ', ' + facility.city;
+
+  return (
+    <styles.Page>
+      <styles.ImageWrapper>
+        <styles.EventImage src={BPLogo} alt="EventImage" />
+        <styles.GradientOverlay />
+      </styles.ImageWrapper>
```
Comment (celinechoiii) [inline app/events/[event_id]/page.tsx:91]: delete these are replace with this:
`<styles.EventImage src={BPLogo} alt="EventImage" />`

File: api/supabase/queries/volunteers.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import { UUID } from 'crypto';
+import supabase from '../createClient';
+
+export async function fetchPerformer(event_id: UUID) {
```
Comment (celinechoiii) [inline api/supabase/queries/volunteers.ts:4]: make this function name a bit more descriptive! something like `fetchPerformerByEventID`

**Comment by celinechoiii** [comment]
to answer your questions:
- it would be nice if you could create a function to format the phone numbers! you can put this in the `utils` folder
- if the `needs_host` boolean is false, the second contact will be the facility host. if true, the volunteer host contact will be listed
- for now, we can assume that there will always be a host or performer who signed up! but that is a good point. i will relay this to jinkang and ask his thoughts

overall, amazing job!! once the changes are made, i should be able to merge!
